# IS 362 Week 5 Assignment

Professor Charles Pak

Angela Hernandez

2/22/24

Your task in this week's assignment is to answer three questions:

1. What is the northernmost airport in the United States?
2. What is the easternmost airport in the United States?
3. On February 12th, 2013, which New York area airport had the windiest weather?



To begin, I'll start by loading the necessary CSV files into pandas to understand the structure of the data, focusing on columns that indicate location for airports and weather conditions for the weather dataset.

For the first two questions, I'll filter and sort the 'airports.csv' data by lat and long to identigy the northernmost and eastermost airports in the US.

For the third question, I'll analyze the 'weather.csv' data to find out which New York area airport had the windiest weather on February 12th, 2013.

In [2]:
import pandas as pd

In [3]:
airports_df = pd.read_csv("C:/Users/Jessie/Documents/IS 362/Week 5/airports.csv")
weather_df = pd.read_csv("C:/Users/Jessie/Documents/IS 362/Week 5/weather.csv")

In [5]:
# Nothernmost Airports Analysis
northernmost_airports = airports_df.sort_values(by='lat', ascending=False).head(5)
print("Northernmost Airports:")
print(northernmost_airports)

Northernmost Airports:
     faa                                        name        lat         lon  \
417  EEN                     Dillant Hopkins Airport  72.270833   42.898333   
230  BRW                  Wiley Post Will Rogers Mem  71.285446 -156.766003   
110  AIN                          Wainwright Airport  70.638056 -159.994722   
708  K03                               Wainwright As  70.613378 -159.860350   
152  ATK  Atqasuk Edward Burnell Sr Memorial Airport  70.467300 -157.436000   

     alt  tz dst              tzone  
417  149  -5   A                NaN  
230   44  -9   A  America/Anchorage  
110   41  -9   A  America/Anchorage  
708   35  -9   A  America/Anchorage  
152   96  -9   A  America/Anchorage  


In [6]:
# Easternmost Airports Analysis
easternmost_airports = airports_df[airports_df['lon'] < 0].sort_values(by='lon', ascending=True).head(5)
print("\nEasternmost Airports:")
print(easternmost_airports)


Easternmost Airports:
      faa              name        lat         lon  alt  tz dst  \
95    ADK      Adak Airport  51.878000 -176.646000   18  -9   A   
112   AKB      Atka Airport  52.220278 -174.206389   56  -9   A   
522   GAM   Gambell Airport  63.766800 -171.733000   27  -9   A   
1281  SVA  Savoonga Airport  63.686400 -170.493000   53  -9   A   
1251  SNP    St Paul Island  57.167333 -170.220444   63  -9   A   

                  tzone  
95    America/Anchorage  
112   America/Anchorage  
522   America/Anchorage  
1281  America/Anchorage  
1251  America/Anchorage  


# Northernmost Airports:

1. Dillant Hopkins Airport (EEN) at 72.270833 latitude.
2. Wiley Post Will Rogers Mem (BRW) at 71.285446 latitude.
3. Wainwright Airport (AIN) at 70.638056.
4. Wainwright AS (K03) at 70.613378.
5. Atqask Edward Burnell Sr Memorial Airport (ATK) at 70.467300 latitude.

# Easternmost Airports

1. Adak Airport (ADK) at -176.646000 longitude.
2. Atka Airport (AKB) at -174.206389 longitude.
3. Gambell Airport (GAM) at -171.733000 longitude.
4. Savoonga Airport (SVA) at -170.493000 longitude.
5. St Paul Island (SNP) at -170.220444 longitude.

Given the outlier in the northenmost airports data (Dillant Hopkins Airport with a latitude that suggests it should not be in the far north), further verification is needed.  For the easternmost airport, we consider the least negative longitude values, recognizing that due to the International Data Line, these locations are technically the easternmost points in the US.

Now I'll proceed to identify the windiest weather on February 12, 2013 at a New York area airport.

In [7]:
# Filter weather data for February 12, 2013 for New York area airports (EWR, JFK, LGA)
ny_weather_feb12 = weather_df[
    (weather_df['year'] == 2013) &
    (weather_df['month'] == 2) &
    (weather_df['day'] == 12) &
    (weather_df['origin'].isin(['EWR', 'JFK', 'LGA']))
]

In [9]:
windiest_weather = ny_weather_feb12.sort_values(by='wind_speed', ascending=False).head(1)
print("\nWindiest Weather on February 12, 2013 at a New York Area Airport:")
print(windiest_weather)


Windiest Weather on February 12, 2013 at a New York Area Airport:
     origin  year  month  day  hour   temp   dewp  humid  wind_dir  \
1009    EWR  2013      2   12     3  39.02  26.96  61.63     260.0   

      wind_speed  wind_gust  precip  pressure  visib             time_hour  
1009  1048.36058        NaN     0.0    1008.3   10.0  2013-02-12T08:00:00Z  


# Conclusion

Summary of findings:

1. Northernmost Airport: Preliminary analysis suggests Wiley Post Will Rogers Memorial Airport (BRW) is the northernmost airport but Dillant Hopkins Airport (EEN) appeared as an outlier likely due to data entry error.
2. Easternmost Airport: Adak Airport (ADK) appears to be the easternmost based on longitude values.
3. Windiest Weather on February 12, 2013: Identified at EWR with a wind speed of 1048.36058 but the wind speed is an outlier, indicating that the data may need to be verified.